redaxo/src/core/lib/console/assets_sync.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+/**
+ * @package redaxo\core
+ * 
+ * @author staabm
+ *
+ * @internal
+ */
+class rex_command_assets_sync extends rex_console_command
+{
+    protected function configure()
+    {
+        $this
+            ->setDescription('Sync folders and files of /assets with /redaxo/src/addons/my-addon/assets (or plugin) respectively /redaxo/src/core/assets folders');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $io = $this->getStyle($input, $output);
+
+        foreach(rex_package::getInstalledPackages() as $package) {
+            if ($package instanceof rex_addon) {
+                $assetsPublicPath = $package->getAssetsPath();
+                $assetsSrcPath = $package->getPath('assets/');
+            } else {
+                // we assume "plugin" assets folder is a subfolder of the "addon" assets folder
+                // and will therefore already be synced when handling the addon
+                continue;
+                // $feAssetsPath = rex_path::pluginAssets($package->getAddon()->getName(), $package->getName());
+                // $beAssetsPath = rex_path::plugin($package->getAddon()->getName(), $package->getName(), 'assets/');
+            }
+
+            // dont create top level "assets" folder when it doesnt exist
+            if (!file_exists($assetsPublicPath)) {
+                $io->comment("skip not existing assets/ folder: $assetsPublicPath");
+                continue;
+            }
+            if (!file_exists($assetsSrcPath)) {
+                $io->comment("skip not existing assets/ folder: $assetsSrcPath");
+                continue;
+            }

```

Comment:
Warum ist dieser Ausschluss sinnvoll?
Wenn ich gerade ein Addon entwickle und erstmals Assets anlege, dann will ich die doch auch synchronisieren an die andere Stelle?

Reply:
Ich sehe diesen folder als „opt-in kennzeichen“. Wenn im addon/plugin-src das verzeichnis nicht existiert dann will man auch keinen sync.

Reply:
Ich finde, den Willen kann man nicht an der Existenz des Ordners festmachen.

Beispiel 1:
src: leer (nicht vorhanden)
fe: enthält `foo.css`

Du gehst nun davon aus, dass foo.css nicht gesynct werden soll.

Beispiel 2:
src: `bar.css`
fe: `foo.css`, `bar.css`

Nun gehst du aber davon aus, dass die foo.css doch gesynct werden soll.

Das ist für mich irgendwie nicht schlüssig, warum in einem Fall der eine Wille angenommen wird, bloß weil bisher noch keine Datei vorhanden, im anderen Fall bei der selben neuen Datei aber ein anderer Wille angenommen wird.